bcf_sr_sort.c

```diff
@@ -0,0 +1,645 @@
+                int k = 1<<kbit;
+                if ( !(i & k) ) continue;
+                if ( max < SR_SCORE(srt,j,k) ) max = SR_SCORE(srt,j,k);
+            }
+        }
+        srt->score[i] = max;
+    }
+}
+static int multi_is_exact(var_t *avar, var_t *bvar)
+{
+    if ( avar->nalt != bvar->nalt ) return 0;
+
+    int alen = strlen(avar->str);
+    int blen = strlen(bvar->str);
+    if ( alen != blen ) return 0;
+
+    char *abeg = avar->str;
+    while ( *abeg )
+    {
+        char *aend = abeg;
+        while ( *aend && *aend!=',' ) aend++;
+
+        char *bbeg = bvar->str;
+        while ( *bbeg )
+        {
+            char *bend = bbeg;
+            while ( *bend && *bend!=',' ) bend++;
+            if ( bend - bbeg == aend - abeg && !strncasecmp(abeg,bbeg,bend-bbeg) ) break;
+            bbeg = *bend ? bend+1 : bend;
+        }
+        if ( !*bbeg ) return 0;
+
+        abeg = *aend ? aend+1 : aend;
+    }
+    return 1;
+}
+static int multi_is_subset(var_t *avar, var_t *bvar)
+{
+    char *abeg = avar->str;
+    while ( *abeg )
+    {
+        char *aend = abeg;
+        while ( *aend && *aend!=',' ) aend++;
+
+        char *bbeg = bvar->str;
+        while ( *bbeg )
+        {
+            char *bend = bbeg;
+            while ( *bend && *bend!=',' ) bend++;
+            if ( bend - bbeg == aend - abeg && !strncasecmp(abeg,bbeg,bend-bbeg) ) return 1;
+            bbeg = *bend ? bend+1 : bend;
+        }
+        abeg = *aend ? aend+1 : aend;
+    }
+    return 0;
+}
+int32_t pairing_score(sr_sort_t *srt, int ivset, int jvset)
+{
+    varset_t *iv = &srt->vset[ivset];
+    varset_t *jv = &srt->vset[jvset];
+
+    // Restrictive logic: the strictest type from a group is selected,
+    // so that, for example, snp+ref does not lead to the inclusion of an indel
+    int i,j;
+    uint32_t min = UINT32_MAX;
+    for (i=0; i<iv->nvar; i++)
+    {
+        var_t *ivar = &srt->var[iv->var[i]];
+        for (j=0; j<jv->nvar; j++)
+        {
+            var_t *jvar = &srt->var[jv->var[j]];
+            if ( srt->pair & BCF_SR_PAIR_EXACT )
+            {
+                if ( ivar->type != jvar->type ) continue;
+                if ( !strcmp(ivar->str,jvar->str) ) return UINT32_MAX;  // exact match, best possibility
+                if ( multi_is_exact(ivar,jvar) ) return UINT32_MAX; // identical alleles
+                continue;
+            }
+            if ( ivar->type==jvar->type && !strcmp(ivar->str,jvar->str) ) return UINT32_MAX;  // exact match, best possibility
+            if ( ivar->type & jvar->type && multi_is_subset(ivar,jvar) ) return UINT32_MAX; // one of the alleles is identical
+
+            uint32_t score = SR_SCORE(srt,ivar->type,jvar->type);
+            if ( !score ) return 0;     // some of the varsets in the two groups are not compatible, will not pair
+            if ( min>score ) min = score;
+        }
+    }
+    if ( srt->pair & BCF_SR_PAIR_EXACT ) return 0;
+
+    assert( min!=UINT32_MAX );
```

Comment:
Under what circumstances could this assert fire?  Would it ever be possible to get through the whole loop without setting `min`?